tests-perf/locust/locust-notifications.py

```diff
@@ -1,24 +1,81 @@
-import os
+import sys, os
+sys.path.append(os.path.abspath(os.path.join("..", "tests_smoke")))
 
+from datetime import datetime
 from dotenv import load_dotenv
 from locust import HttpUser, constant_pacing, task
+from tests_smoke.smoke.common import job_line, rows_to_csv
 
 load_dotenv()
-AUTH_HEADER = os.getenv("TEST_AUTH_HEADER")
 
 
 class NotifyApiUser(HttpUser):
 
     wait_time = constant_pacing(60)
-    host = "https://api.staging.notification.cdssandbox.xyz"
+    host = os.getenv("LOAD_TEST_DOMAIN", "https://api.staging.notification.cdssandbox.xyz")
 
-    @task
-    def send_notifications(self):
-        headers = {"Authorization": AUTH_HEADER}
+    def __init__(self, *args, **kwargs):
+        super(NotifyApiUser, self).__init__(*args, **kwargs)
+
+        self.headers = {"Authorization": os.getenv("TEST_AUTH_HEADER")}
+        self.email = "success@simulator.amazonses.com"
+        self.template_id = "5ebee3b7-63c0-4052-a8cb-387b818df627"
+
+    @task(1)
+    def send_email_notifications(self):
+        json = self.__email_json("a59b313d-8de2-4973-ac2f-66de7ec0b239")
+
+        self.client.post("/v2/notifications/email", json=json, headers=self.headers)
+
+    @task(2)
+    def send_email_with_attachment_notifications(self):
+        personalisation = {
+            "application_file": {
+                "file": "Q29udGVudCBvZiBBdHRhY2hlZCBmaWxl",
+                "filename": "attached_file.txt",
+                "sending_method": "attach",
+            }
+        }
+        json = self.__email_json(self.template_id, personalisation)
+
+        self.client.post("/v2/notifications/email", json=json, headers=self.headers)
+
+    @task(4)
+    def send_email_with_link_notifications(self):
+        personalisation = {
+            "application_file": {
+                "file": "Q29udGVudCBvZiBBdHRhY2hlZCBmaWxl",
+                "filename": "attached_file.txt",
+                "sending_method": "link",
+            }
+        }
+        json = self.__email_json(self.template_id, personalisation)
+
+
+        self.client.post("/v2/notifications/email", json=json, headers=self.headers)
+
+    @task(8)
+    def send_bulk_notifications(self):
+        json = {
+            "name": f"My bulk name {datetime.utcnow().isoformat()}",
+            "template_id": self.template_id,
+            "csv": rows_to_csv([["email address", "application_file"], *job_line(self.email, 2)])
+        }
+
+        self.client.post("/v2/notifications/bulk", json=json, headers=self.headers)
+
+    @task(16)
+    def send_sms_notifications(self):
```

Comment:
Do we need to load test sending sms on a regular basis? They cost $0.01 each, and sending thousands a day will add up.

Reply:
Hello Steve, I gathered that using this number 12897682684 is free

Reply:
no, it still costs to send to it.

Reply:
We can use a number as listed [in the smoke testing section of the documentation](https://documentation.notification.canada.ca/en/testing.html#smoke-testing), that won't really get sent. One could argue that we should test the whole execution path but it's the most we can do without incurring any cost.

Reply:
Bump here, is this discussion considered resolved? Are we okay for testing with SMS using the test phone numbers?

Reply:
The test # won't cost us anything, though at the same time they won't create a notification and test the backend. IMO we should create real Notify test numbers / addresses that create a notification and do everything except the SES / SMS send (at that point the notifications will be just marked as delivered or failed as desired). This will allow us and our users to fully test our / their systems. This is out of scope of this PR though.

Reply:
We discussed this offline. The simulated emails will test throughout AWS backend but the simulated SMS will simply stop at the API. It would be nice to have the SMS to go deeper into the pipeline to have a better performance tests on these but that would requires further changes into Notify. @sastels [created this task](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/36) to capture necessary changes.